localstack/services/events/target.py

```diff
@@ -0,0 +1,303 @@
+import json
+import logging
+import uuid
+from abc import ABC, abstractmethod
+
+from botocore.client import BaseClient
+
+from localstack.aws.api.events import (
+    Arn,
+    Target,
+)
+from localstack.aws.connect import connect_to
+from localstack.utils import collections
+from localstack.utils.aws.arns import (
+    extract_service_from_arn,
+    firehose_name,
+    sqs_queue_url_for_arn,
+)
+from localstack.utils.aws.client_types import ServicePrincipal
+from localstack.utils.strings import to_bytes
+from localstack.utils.time import now_utc
+
+LOG = logging.getLogger(__name__)
+
+
+class TargetSender(ABC):
+    def __init__(
+        self,
+        target: Target,
+        region: str,
+        account_id: str,
+        rule_arn: Arn,
+        service: str,
+    ):
+        self.target = target
+        self.region = region
+        self.account_id = account_id
+        self.rule_arn = rule_arn
+        self.service = service
+
+        self._validate_input(target)
+        self._client: BaseClient | None = None
+
+    @property
+    def arn(self):
+        return self.target["Arn"]
+
+    @property
+    def client(self):
+        """Lazy initialization of internal botoclient factory."""
+        if self._client is None:
+            self._client = self._initialize_client()
+        return self._client
+
+    @abstractmethod
+    def send_event(self):
+        pass
+
+    def _validate_input(self, target: Target):
+        """Provide a default implementation that does nothing if no specific validation is needed."""
+        # TODO add For Lambda and Amazon SNS resources, EventBridge relies on resource-based policies.
+        pass
+
+    def _initialize_client(self) -> BaseClient:
+        """Initializes internal botocore client.
+        If a role from a target is provided, the client will be initialized with the assumed role.
+        If no role is provided the client will be initialized with the account ID and region.
```

Comment:
In which cases does this apply? Isn't a role ARN mandatory for EventBridge?

Reply:
role arn is only required for EC2 instances, Kinesis Data Streams, Step Functions state machines and API Gateway APIs